assets/aml-benchmark/components/src/dataset_downloader/main.py

```diff
@@ -0,0 +1,260 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""Entry script for Dataset Downloader Component."""
+
+import argparse
+import os
+from typing import Optional, List
+from concurrent.futures import ProcessPoolExecutor
+
+from datasets import load_dataset, get_dataset_split_names, get_dataset_config_names
+import pandas as pd
+from azureml._common._error_definition.azureml_error import AzureMLError
+
+from utils.helper import get_logger, log_mlflow_params
+from utils.exceptions import (
+    swallow_all_exceptions,
+    BenchmarkValidationException,
+    DatasetDownloadException,
+)
+from utils.error_definitions import BenchmarkValidationError, DatasetDownloadError
+
+
+logger = get_logger(__name__)
+ALL = "all"
+
+
+def parse_args() -> argparse.Namespace:
+    """Parse command-line arguments."""
+    parser = argparse.ArgumentParser(description=f"{__file__}")
+    parser.add_argument(
+        "--dataset_name",
+        type=str,
+        default=None,
+        help="Name of the dataset to download.",
+    )
+    parser.add_argument(
+        "--configuration",
+        type=str,
+        default=None,
+        help="Configuration of the dataset to download.",
+    )
+    parser.add_argument(
+        "--split",
+        type=str,
+        default=None,
+        help="Split of the dataset to download.",
```

Comment:
Please include information about allowed values.

Reply:
there is no restriction on the allowed values as it can be any str, have added the details about what value `all` can do.